tests/test_rdscluster.py

```diff
@@ -736,6 +736,25 @@ def test_clear_permissions(self):
         )
         self.assertEqual(len(restore_permissions_after), 0)
 
+    def test_pending_maintenance(self):
+        session_factory = self.record_flight_data("test_rdscluster_pending_maintenance")
```

Comment:
@sgungi-intuit looks like tests are failing

This line should be verified locally then updated to "replay_flight_data"